packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts

```
@@ -618,10 +618,18 @@ export default class FoxgloveWebSocketPlayer implements Player {
           : new JsonMessageWriter();
 
         // Add type definitions for service response and request
+        let datatypesNeedsUpdate = false;
         for (const [name, types] of [...parsedRequest.datatypes, ...parsedResponse.datatypes]) {
+          if (this.#datatypes.has(name)) {
+            // Assumption: Schemas do not change.
+            continue;
```

Do we need to make this assumption? Can we verify the assumption and raise an error to the user if it doesn't hold instead of silently failing?